morello/src/spec.rs

```diff
@@ -1624,18 +1624,34 @@ pub mod macros {
         }
     }
 
+    #[macro_export]
+    macro_rules! dimsize {
+        (0) => {
+            compile_error!("0 is not allowed for DimSize.")
+        };
+        ($dim:literal) => {{
+            const _: () = {
+                let _x: u32 = $dim;
+            };
+            unsafe { std::num::NonZeroU32::new_unchecked($dim) }
```

Comment:
@samkaufman   If we want to use `nonzero!`, which internally uses unsafe code, we need to add [paste crate](https://crates.io/crates/paste) to add the `u32` suffix so that `nonzero!` can correctly parse the literal.  If that's not a big dead for you, I can switch to it.